Experimental/MultiplayerUseCases/Assets/Common/Scripts/NetworkManagerUI.cs

```diff
@@ -1,71 +1,102 @@
+            m_DisconnectButton = UIElementsUtils.SetupButton("DisconnectButton", Disconnect, false, m_Root, "Disconnect", "Disconnects participant from session");
+            UIElementsUtils.SetupButton("QuitSceneButton", QuitScene, true, m_Root, "Quit Scene", "Quits scene and brings you back to the scene selection screen");
         }
 
         void Start()
         {
-            SetButtonStateAndColor(m_DisconnectButton, false, false);
             m_ServerOnlyOverlay.gameObject.SetActive(false);
         }
 
         void StartServer()
         {
+            SetNetworkPortAndAddress(ushort.Parse(m_PortInputField.value), m_AddressInputField.value, k_DefaultServerListenAddress);
+            NetworkManager.Singleton.GetComponent<UnityTransport>().ConnectionData.ServerListenAddress = m_AddressInputField.value;
             NetworkManager.Singleton.StartServer();
-            EnableAndHighlightButtons(m_ServerButton, false);
-            SetButtonStateAndColor(m_DisconnectButton, false, true);
             m_ServerOnlyOverlay.gameObject.SetActive(true);
+
+            Button[] buttons = new Button[] {m_ClientButton, m_HostButton};
+            EnableButtons(buttons, false);
+            m_DisconnectButton.SetEnabled(true);
+            m_ServerVisualHighlight.visible = true;
         }
 
         void StartHost()
         {
+            SetNetworkPortAndAddress(ushort.Parse(m_PortInputField.value), m_AddressInputField.value, k_DefaultServerListenAddress);
             NetworkManager.Singleton.StartHost();
-            EnableAndHighlightButtons(m_HostButton, false);
-            SetButtonStateAndColor(m_DisconnectButton, false, true);
+            Button[] buttons = new Button[] { m_ClientButton, m_ServerButton };
+            EnableButtons(buttons, false);
+            m_DisconnectButton.SetEnabled(true);
+            m_HostVisualHighlight.visible = true;
         }
 
         void StartClient()
         {
+            SetNetworkPortAndAddress(ushort.Parse(m_PortInputField.value), m_AddressInputField.value, k_DefaultServerListenAddress);
             NetworkManager.Singleton.StartClient();
-            EnableAndHighlightButtons(m_ClientButton, false);
-            SetButtonStateAndColor(m_DisconnectButton, false, true);
+            Button[ ] buttons = new Button[]{m_HostButton, m_ServerButton};
+            EnableButtons(buttons, false);
+            m_DisconnectButton.SetEnabled(true);
+            m_ClientVisualHighlight.visible = true;
         }
 
         void Disconnect()
         {
             NetworkManager.Singleton.Shutdown();
-            EnableAndHighlightButtons(null, true);
-            SetButtonStateAndColor(m_DisconnectButton, false, false);
             m_ServerOnlyOverlay.gameObject.SetActive(false);
+            m_ServerVisualHighlight.visible = false;
+            m_HostVisualHighlight.visible = false;
+            m_ClientVisualHighlight.visible = false;
```

Comment:
1. This code should not be in a `Disconnect()` method. That's why we had `EnableAndHighlightButtons()` I don't understand why that method was removed instead of edited to accomodate the UGUI to UITK changes, only to put its code in the other various methods.
2. Nitpick: There's no need for 3 different sprites: you could just have one and reparent it within the document. It's more scalable, and requires you to manage less things.